pkg/cloud/tags/tags.go

```diff
@@ -78,6 +78,9 @@ func (b *Builder) Apply() error {
 
 // Ensure applies the tags if the current tags differ from the params.
 func (b *Builder) Ensure(current infrav1.Tags) error {
+	if b.params == nil {
```

Comment:
This check is already in `Apply()`.
`Ensure()` does not need to have this check as it checks for diff. Having a nil params will cause diff to be empty and return with success (this is not valid use case, we always have tags but still, I don't think this check is the responsibility of this method)

Reply:
The reason this check has been added here, if you see this line, it uses `*b.params` as input to `computeDiff()`, and if we pass `b.params` as `nil`, we will get panic here, hence to avoid this, `b.params` should be check first, before computing the diff

Reply:
Makes sense, did not see that we were dereferencing it.

Reply:
Done